public/extensions/iframe.js

```diff
@@ -0,0 +1,156 @@
+import daily from "./core.js";
+import flexpanel from "./flexpanel.js";
+import callstate from "./callstate.js";
+
+let fp, iframeEl, locationEl, call;
+let open = false;
+
+let defaultButtons = {
+  selectUrl: {
+    iconPath: "https://www.svgrepo.com/show/107601/globe.svg",
+    label: "Iframe",
+    tooltip: "Open Iframe",
+  },
+};
+
+// configure defaults that can be
+// overridden by calling configure()
+let props = {
+  url: "about:blank",
+  buttons: defaultButtons,
+  showUrl: false,
+};
+
+// overloading the "selectUrl" button to work around a bug with updateCustomTrayButtons for now
+function handleSelectButton() {
+  if (callstate.state.iframe?.open === true) {
+    callstate.updateCallState("iframe", { open: false });
+  } else {
+    selectUrl();
+  }
+}
+
+let self;
+export default self = {
+  configure: function (p) {
+    Object.assign(props, p);
+  },
+  open: function (url) {
+    // sets a new URL for the iframe and opens
+    props.url = url;
+    open = true;
+    callstate.updateCallState("iframe", { url, open });
+  },
+  close: function () {
+    // closes the iframe and clears URL
+    props.url = "about:blank";
+    open = false;
+    callstate.updateCallState("iframe", { url: props.url, open });
+  },
+  show: function () {
+    // just shows the iframe (flexpanel)
+    open = true;
+    callstate.updateCallState("iframe", { open });
+  },
+  hide: function () {
+    // just hides without removing URL
+    open = false;
+    callstate.updateCallState("iframe", { open });
+  },
+  setUrl: function (url) {
+    // updates URL without changing
+    // show/hide state
+    props.url = url;
+    callstate.updateCallState("iframe", { url: props.url });
+  },
+};
+
+function selectUrl() {
+  let url = prompt("Enter a URL", props.url);
+  if (url) {
+    self.open(url);
+  }
+}
+
+function handleUrlUpdate(url) {
+  props.url = url;
+  locationEl.innerHTML = url;
+  iframeEl.src = url;
+}
+
+function handleShow() {
+  flexpanel.open();
+  open = true;
+  // call.updateCustomTrayButtons({ close: closeButton, temp: tempButton });
```

Comment:
Can these (And LN:90) go away? Or if they're informational, can we add a comment above?